addons/website_sale/controllers/main.py

```diff
@@ -285,6 +285,9 @@ def shop(self, page=0, category=None, search='', min_price=0.0, max_price=0.0, p
             limit=None, order=self._get_search_order(post), options=options)
         search_product = details[0].get('results', request.env['product.template']).with_context(bin_size=True)
 
+        if attrib_set:
+            search_product = search_product.filtered(lambda product: product._get_possible_combination_count(list(attrib_set)))
```

Comment:
This is too expensive (looped search with already complicated stuff)..

I propose something along those lines (Untested):
```suggestion
            attrib_ids = self.env['product.attribute'].browse(attrib_ids)
            search_product = search_product.filtered(lambda tmpl:\
                any(attrib_ids <= p.product_template_attribute_value_ids.attribute_id for p in tmpl.product_variant_ids))
```

It is still pretty expensive but I don't see how we can do it better.
`product_variant_ids` will contain only the activated variants (excluding archived and `exclude for` variants).
And we use set comparison to check if a variant with the filtered attributes exists, which should be pretty fast.

Let me know if there are issues to make it work.

Reply:
what about the dynamically created variants?